### DISCUSSION TITLE
How to deal with compatibility issues with TongWeb

### DISCUSSION THREAD

**Discussion by liunianduanzhinian** [apache/skywalking#12623] [Q&A] [answered]
TongWeb is a non open source middleware in China，the trace function can only trace a small number of interfaces.Is there any mature solution to adapt to non open source middleware？

**Comment by wu-sheng** [comment, marked as answer]
`non open source middleware` means the source codes are not available. So, there is nothing we can do. You could seek commercial services from TongWeb company or SkyWalking's distribution company.

But I don't think upstream has anything to do with that.

**Reply by liunianduanzhinian** [reply]
Thank you for your answer!
